src/main.js

```diff
@@ -95,19 +96,113 @@ buttons.forEach((button) => {
   });
 });
 
-//shrink header on scroll
+// Object Literal Module
+
+// (function () {
+//   const buttons = {
+//     init: function () {
+//       this.cacheDom();
+//       this.bindEvents();
+//       // this.render();
+//     },
+
+//     cacheDom: function () {
+//       this.buttons = Array.from(document.querySelector(".buttons").children);
+//       this.visited = this.buttons[0];
+//       this.bucketlist = this.buttons[1];
+//       this.all = this.buttons[2];
+//       this.gallery = document.querySelector(".gallery");
+//     },
+
+//     bindEvents: function () {
+//       this.visited.addEventListener("click", addVisited);
+//       this.bucketlist.addEventListener("click", addBucketlist);
+//       this.all.addEventListener("click", showAll);
+//     },
+
+//     addVisited: function () {
+//       this.gallery.classList.add("visited");
+//       this.gallery.classList.remove("bucketlist");
+//     },
+
+//     addBucketlist: function () {
+//       this.gallery.classList.add("bucketlist");
+//       this.gallery.classList.remove("visited");
+//     },
+//     showAll: function () {
+//       this.gallery.classList.remove("visited");
+//       this.gallery.classList.remove("bucketlist");
+//     },
+//   };
+
+//   buttons.init();
+// })();
 
-const navbar = document.querySelector(".navbar");
+//shrink header on scroll
 
-document.addEventListener("scroll", () => {
+const shrinkNav = document.addEventListener("scroll", () => {
+  const navbar = document.querySelector(".navbar");
   navbar.classList.add("shrink");
 });
 
 //Change header behaviour based on scrolling position
-let scrollPos = 0;
-
-window.addEventListener("scroll", function () {
+const navSize = window.addEventListener("scroll", function () {
+  const scrollPos = 0;
   if (document.body.getBoundingClientRect().top === scrollPos) {
     document.querySelector(".navbar").classList.remove("shrink");
   }
 });
+
+//create new city
+const addCity = function () {
```

Comment:
In JS we can create as many modules as needed. I recommend you to create a module for the Form too. You can name modules (and init them) like this: (based on my previous comment)

```
const moduleDestinationsSection = function () {
  let buttons;
  // ....

  // Do not execute "init", instead return it, so that the module can be initiated from the outside
  return {
    init,
  };
}

// Initiate the module
moduleDestinationsSection().init();
```

Reply:
A next step: You'll notice the code is already growing like crazy. You can split each module per file to help you keeping the focus in each file.

```js
// src/scripts/modules/destinations.js

// You don't need to create the scoped function because it's already "scoped" in its own file.
let buttons; 
// ...

// every single function of this module....

// and then export (instead of return) the last function to be used from the outside

export init
```

and then in `main.js`

```js
import moduleDestinations from "./scripts/destinations.js";
// ...

// Init the module!
moduleDestinations.init()
```

Reply:
```// You don't need to create the scoped function because it's already "scoped" in its own file.```

This means I don't need ```const moduleDestinations = (function () {})(), so in ```import moduleDestinations from "./scripts/modules/destinations.js";``` I'm giving it a name?

Reply:
I'm getting an error, ```Uncaught ReferenceError: destinationsList is not defined```

Reply:
> This means I don't need const moduleDestinations = (function () {})(), so in import moduleDestinations from "./scripts/modules/destinations.js";``` I'm giving it a name?

Yes, exactly. One of the reason for us to create that [IIFE](https://flaviocopes.com/javascript-iife/) in first place is to avoid scoped issues and conflict with external variables. By doing the module in a new file the conflicts don't exist anymore, so you can just declare the module content directly :)

Just one last thing, I edited how the export should be done, please take a look again.

Regarding the error, you need to import that file in the new module file too. Everytime you need some external to a file, you need to import it at the top :)

Reply:
So I added ``` import destinationsList from "./scripts/destinationList.js";``` at the top of destinations.js and I'm still getting a 'can't resolve dependency' error.

Reply:
I don't understand this syntax ```const module = { init };```, this is init() that I'm exporting?

Reply:
so we build a module and define what can be exported from it. It's the same as

```
const module = {
  init: init,
}
```

We could also do in other way


```js
export function init() {...}

// index.js
import { init } from "./scripts/modules/destinations.js";

init()
```

or

```js
export function init() {...}

// index.js
import * as moduleDestinations from "./scripts/modules/destinations.js";

moduleDestinations.init()
```

You can read more about [import and export in JS](https://javascript.info/import-export)